src/providers/index.ts

```diff
@@ -242,3 +241,32 @@ export async function loadApiProviders(
   }
   throw new Error('Invalid providers list');
 }
+
+/**
+ * Get the IDs of the providers in the providerPaths list.
+ * @param providerPaths - The list of providers to get the IDs of.
+ * @returns The IDs of the providers in the providerPaths list.
+ */
+export function getProviderIds(providerPaths: TestSuiteConfig['providers']): string[] {
+  if (typeof providerPaths === 'string') {
+    return [providerPaths];
+  } else if (typeof providerPaths === 'function') {
+    return ['custom-function'];
+  } else if (Array.isArray(providerPaths)) {
+    return providerPaths.map((provider, idx) => {
+      if (typeof provider === 'string') {
+        return provider;
+      }
+      if (typeof provider === 'function') {
+        return provider.label || `custom-function-${idx}`;
+      }
+      if ((provider as ProviderOptions).id) {
+        return (provider as ProviderOptions).id!;
+      }
+      const id = Object.keys(provider)[0];
+      const providerObject = (provider as ProviderOptionsMap)[id];
+      return providerObject.id || id;
+    });
```

Comment:
idea: if you extract the logic of taking a single providerPath and returning the id, then use that as a subroutine for this, making the input an array if isn't, then you can avoid repeating the logic inside the map

Reply:
I also thought about this; decided to mirror `loadApiProvider` but IMO they should both be refactored to do what you're suggesting.